### PR TITLE
[Metrol-261]

### PR DESCRIPTION
**Pull request by wiswis15**

### CODE REVIEW

File: Source/core/IPCConnector.h [hunk 1]
```diff
@@ -331,17 +331,16 @@ namespace Core {
             // -----------------------------------------------------
             HAS_MEMBER(Clear, hasClear);
 
-            typedef hasClear<PACKAGE, void (PACKAGE::*)()> TraitClear;
 
-            template <typename SUBJECT, const uint32_t ID>
-            inline typename Core::TypeTraits::enable_if<RawSerializedType<SUBJECT, ID>::TraitClear::value, void>::type
+            template <typename SUBJECT=PACKAGE, const uint32_t ID>
```
Comment (MFransen69) [inline Source/core/IPCConnector.h]: Same as the remarks above. I propose before I review further that you check all the changes for this. Okay?

**Review by MFransen69 — CHANGES_REQUESTED** [review]

File: Source/core/IPCChannel.h [hunk 2]
```diff
@@ -398,17 +398,15 @@ namespace Core {
 
         HAS_MEMBER(InvokeAllowed, hasInvokeAllowed);
 
-        typedef hasInvokeAllowed<EXTENSION, ProxyType<IIPC> (EXTENSION::*)(const ProxyType<IIPC>&) const> TraitInvokeAllowed;
-
-        template <typename A, const bool B>
-        inline typename Core::TypeTraits::enable_if<IPCChannelServerType<A, B>::TraitInvokeAllowed::value, ProxyType<IIPC>>::type
+        template <typename className = EXTENSION, const bool B>
```
[the remark this answers is not in the document]
Comment (wiswis15) [reply Source/core/IPCChannel.h]: done

File: Source/core/IPCChannel.h [hunk 3]
```diff
@@ -398,17 +398,15 @@ namespace Core {
...
+        inline typename Core::TypeTraits::enable_if<hasInvokeAllowed<className, ProxyType<IIPC> (className::*)(const ProxyType<IIPC>&) const>::value, ProxyType<IIPC>>::type
         __InvokeAllowed(const Client& client, const ProxyType<IIPC>& command) const
```
[the remark this answers is not in the document]
Comment (wiswis15) [reply Source/core/IPCChannel.h:403]: done

File: Source/core/IPCConnector.h [hunk 4]
```diff
@@ -331,59 +331,55 @@ namespace Core {
...
+            inline typename Core::TypeTraits::enable_if<!hasClear<SUBJECT, void (SUBJECT::*)()>::value, void>::type
             __Clear()
             {
             }
 
             // -----------------------------------------------------
             // Search for custom handling, Compile time !!!
             // -----------------------------------------------------
-            HAS_MEMBER(Length, hasLength);
+            HAS_MEMBER_NAME(Length, hasLength);
```
Comment (MFransen69) [inline Source/core/IPCConnector.h]: Hi Wissem, why did you now change this to HAS_MEMBER_NAME? Didn't it compile with HAS_MEMBER?

**Review by MFransen69 — CHANGES_REQUESTED** [review]

File: Source/core/IPCConnector.h (hunk 4 above)
Comment (wiswis15) [reply Source/core/IPCConnector.h]: good catch. Sorry, I was testing with it. and committed it by mistake.reverted.

File: Source/core/IPCChannel.h [hunk 5]
```diff
@@ -393,22 +393,20 @@ namespace Core {
...
+        template <typename className = EXTENSION>
+        inline typename Core::TypeTraits::enable_if<hasInvokeAllowed<className, ProxyType<IIPC> (className::*)(const ProxyType<IIPC>&) const>::value, ProxyType<IIPC>>::type
         __InvokeAllowed(const Client& client, const ProxyType<IIPC>& command) const
         {
             return (client.Extension().InvokeAllowed(command));
         }
 
-        template <typename A, const bool B>
-        inline typename Core::TypeTraits::enable_if<!IPCChannelServerType<A, B>::TraitInvokeAllowed::value, ProxyType<IIPC>>::type
+        template <typename className = EXTENSION>
```
Comment (MFransen69) [inline Source/core/IPCChannel.h]: Please use uppercase for template parameters

File: Source/core/Proxy.h [hunk 6]
```diff
@@ -185,17 +185,15 @@ namespace Core {
         // -----------------------------------------------------
         HAS_MEMBER(Initialize, hasInitialize);
 
-        typedef hasInitialize<CONTEXT, uint32_t (CONTEXT::*)()> TraitInitialize;
-
-        template <typename TYPE>
-        inline typename Core::TypeTraits::enable_if<ProxyService<TYPE>::TraitInitialize::value, uint32_t>::type
+        template <typename TYPE=CONTEXT>
+        inline typename Core::TypeTraits::enable_if<hasInitialize<TYPE, uint32_t (TYPE::*)()>::value, uint32_t>::type
         __Initialize()
```
Comment (MFransen69) [inline Source/core/Proxy.h:190]: Please also check if the usage of these methods now should not be changed as well (template params removed). e.g. line 58, 99, 103 etc.)

File: Source/core/Proxy.h [hunk 7]
```diff
@@ -1032,7 +1029,7 @@ namespace Core {
                     ProxyObjectType* baseElement(const_cast<ProxyObjectType*>(this));
 
                     baseElement->__Relinquish<ELEMENT>();
```
Comment (MFransen69) [inline Source/core/Proxy.h]: Think this line should be changed as well but please check all usages of all HAS_MEMBER methods in this file as mentioned above

File: Source/core/Proxy.h [hunk 8]
```diff
@@ -1094,17 +1087,15 @@ namespace Core {
...
-            template <typename TYPE>
-            inline typename Core::TypeTraits::enable_if<ProxyObjectType<TYPE>::TraitRelinquish::value, void>::type
+            template <typename TYPE=ELEMENT>
+            inline typename Core::TypeTraits::enable_if<hasRelinquish<TYPE, void (TYPE::*)()>::value, void>::type
             __Relinquish()
             {
                 ELEMENT::Relinquish();
             }
 
             template <typename TYPE>
```
Comment (MFransen69) [inline Source/core/Proxy.h]: You forgot the default argument, You wiill notice any when you correct the usage and remove the template param from the call :)

File: Source/core/ResourceMonitor.h [hunk 9]
```diff
@@ -258,14 +258,14 @@ namespace Core {
     private:
         HAS_MEMBER(Arm, hasArm);
 
-        template <typename TYPE>
+        template <typename TYPE=WATCHDOG>
         inline typename Core::TypeTraits::enable_if<hasArm<TYPE, void (TYPE::*)()>::value, void>::type
         Arm()
```
Comment (MFransen69) [inline Source/core/ResourceMonitor.h:263]: Please now also adapt the usage of this method and remove the template parameters

File: Source/core/SocketServer.h [hunk 10]
```diff
@@ -306,17 +306,16 @@ namespace Core {
             // -----------------------------------------------------
             HAS_MEMBER(Id, hasId);
 
-            typedef hasId<HANDLECLIENT, void (HANDLECLIENT::*)(uint32_t)> TraitId;
 
-            template <typename SUBJECT>
-            inline typename Core::TypeTraits::enable_if<SocketHandler<SUBJECT>::TraitId::value, void>::type
+            template <typename SUBJECT=HANDLECLIENT>
+            inline typename Core::TypeTraits::enable_if<hasId<SUBJECT, void (SUBJECT::*)(uint32_t)>::value, void>::type
             __Id(HANDLECLIENT& object, const uint32_t id)
```
Comment (MFransen69) [inline Source/core/SocketServer.h:312]: as above

File: Source/websocket/WebLink.h [hunk 11]
```diff
@@ -323,32 +323,29 @@ namespace Web {
...
+        ReceiveData( uint8_t* dataFrame, const uint16_t receivedSize)
         {
             return (_deserialiserImpl.Deserialize(dataFrame, receivedSize));
         }
 
-        template <typename CLASSNAME>
-        inline typename Core::TypeTraits::enable_if<CLASSNAME::TraitSerializer::value, uint16_t>::type
+        template <typename CLASSNAME=TRANSFORM>
+        inline typename Core::TypeTraits::enable_if<hasTransform<CLASSNAME, uint16_t (CLASSNAME::*)(BaseSerializer&, uint8_t* data, const uint16_t maxSize)>::value, uint16_t>::type
         SendData(const CLASSNAME&, uint8_t* dataFrame, const uint16_t receivedSize)
```
Comment (MFransen69) [inline Source/websocket/WebLink.h]: Think the first param can be removed as well now.

**Review by MFransen69 — CHANGES_REQUESTED** [review]

File: Source/websocket/WebLink.h (hunk 11 above)
Comment (wiswis15) [reply Source/websocket/WebLink.h:342]: removed it a s well from the has_Transform .

**Review by MFransen69 — APPROVED** [review]

**Pull request closed without merging**